src/main/java/com/gmail/goosius/siegewar/listeners/SiegeWarTownyEventListener.java

```diff
@@ -160,13 +160,13 @@ public void onBlockExploding(TownyExplodingBlocksEvent event) throws TownyExcept
      *
      * @return filtered list
      */
-    private static List<Block> filterExplodeListByTrapWarfareMitigation(List<Block> givenExplodeList) {       
+    private static List<Block> filterExplodeListByTrapWarfareMitigation(List<Block> givenExplodeList) {
         if(!SiegeWarSettings.isTrapWarfareMitigationEnabled())
             return givenExplodeList;
-
+        boolean capZoneMitigationOnly = SiegeWarSettings.isTrapWarfareMitigationCapZoneOnly();
         List<Block> filteredList = new ArrayList<>(givenExplodeList);
         for(Block block: givenExplodeList) {
-            if (SiegeWarDistanceUtil.isLocationInSiegeZoneWildernessAndBelowSiegeBannerAltitude(block.getLocation())) {
+            if(SiegeWarDistanceUtil.isLocationInSiegeZoneWildernessAndBelowSiegeBannerAltitude(block.getLocation(), capZoneMitigationOnly)) {

```

Comment:
```suggestion
            if(SiegeWarDistanceUtil.isLocationInSiegeZoneWildernessAndBelowSiegeBannerAltitude(block.getLocation(), SiegeWarSettings.isTrapWarfareMitigationCapZoneOnly())) {

```

Reply:
- I stored the config value locally as an optimization, as this operation is called for every exploding block in the list.

Reply:
Does that make sense to add that little optimization?